src/formatDate.js

```diff
@@ -51,6 +51,32 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const [,,, oldSeparator] = fromFormat;
+  const [,,, newSeparator] = toFormat;
+
+  const arrDate = date.split(oldSeparator);
+  const dateObj = {};
+  const newDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = arrDate[i];
+  }
+
+  if (dateObj['YY']) {
+    if (dateObj['YY'] < 30) {
+      dateObj['YYYY'] = 20 + dateObj['YY'];
+    } else {
+      dateObj['YYYY'] = 19 + dateObj['YY'];
+    }
+  } else {
+    dateObj['YY'] = dateObj['YYYY'].slice(2);
+  }
+
+  for (let j = 0; j < 3; j++) {
```

Comment:
`3` looks like a magic number move it to the variable or use toFormat.length - 1